From f6ec40aeab76b44285f6348e748e9c4f44474d19 Mon Sep 17 00:00:00 2001
From: Rob Browning <rlb@defaultvalue.org>
Date: Tue, 25 Oct 2011 19:52:51 -0500
Subject: Use CRT_DIR to find crt*.o on ppc64.

The Emacs build should now look for crt*.o in the correct multiarch
location during ppc64 builds.

Provided-by: Hiroyuki Yamamoto <yama1066@gmail.com>
Date: Wed, 26 Oct 2011 01:39:35 +0900
Applied-by: Rob Browning <rlb@defaultvalue.org>
Bug-Debian: http://bugs.debian.org/646609
---
 configure.in   |    8 ++++----
 src/m/macppc.h |    4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/configure.in b/configure.in
index 94efa35..f544aca 100644
--- a/configure.in
+++ b/configure.in
@@ -172,7 +172,7 @@ fi
 CRT_DIR=
 AC_ARG_WITH([crt-dir],dnl
 [AS_HELP_STRING([--with-crt-dir=DIR],[directory containing crtn.o etc.
-This option is only used on x86-64 and s390x GNU/Linux architectures.])])
+This option is only used on x86-64, powerpc64 and s390x GNU/Linux architectures.])])
 CRT_DIR="${with_crt_dir}"
 
 AC_ARG_WITH([gnustep-conf],dnl
@@ -1006,11 +1006,11 @@ AC_CONFIG_LIBOBJ_DIR(src)
 dnl Do this early because it can frob feature test macros for Unix-98 &c.
 AC_SYS_LARGEFILE
 
-## Note: at present CRT_DIR is only used for amdx86-64 and ibms390x.
+## Note: at present CRT_DIR is only used for amdx86-64, powerpc64 and ibms390x.
 ## Other machine types hard-code the location in src/[ms]/*.h.
 case "${canonical}" in
-  x86_64-*-linux-gnu* | s390x-*-linux-gnu* )
-   ## On x86-64 and s390x GNU/Linux distributions, the standard library
+  x86_64-*-linux-gnu* | powerpc64-*-linux-gnu* | s390x-*-linux-gnu* )
+   ## On x86-64, powerpc64 and s390x GNU/Linux distributions, the standard library
    ## can be in a variety of places.  We only try /usr/lib64 and /usr/lib.
    ## For anything else (eg /usr/lib32), it is up the user to specify
    ## the location (bug#5655).
diff --git a/src/m/macppc.h b/src/m/macppc.h
index c0fcae2..78ca68e 100644
--- a/src/m/macppc.h
+++ b/src/m/macppc.h
@@ -50,9 +50,9 @@ along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.  */
 #define LD_SWITCH_MACHINE_TEMACS -Xlinker -znocombreloc
 #ifdef _ARCH_PPC64
 #undef START_FILES
-#define START_FILES pre-crt0.o /usr/lib64/crt1.o /usr/lib64/crti.o
+#define START_FILES pre-crt0.o $(CRT_DIR)/crt1.o $(CRT_DIR)/crti.o
 #undef LIB_STANDARD
-#define LIB_STANDARD -lgcc -lc -lgcc /usr/lib64/crtn.o
+#define LIB_STANDARD -lgcc -lc -lgcc $(CRT_DIR)/crtn.o
 #endif
 #endif
 
